### PR TITLE
doc: update readme

### PR DESCRIPTION
**Pull request by RomainGoussault**
Signed-off-by: Romain Goussault <romain.goussault@owkin.com>

## Related issue

`#` followed by the number of the issue

## Please check if the PR fulfills these requirements

- [ ] If necessary, the [changelog](https://github.com/Substra/substra/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification
- For any breaking changes, companion PRs have been opened on the following repositories:
  - [ ] [substra-tests](https://github.com/Substra/substra)
  - [ ] [substrafl](https://github.com/Substra/substrafl)
  - [ ] [substra-documentation](https://github.com/Substra/substra-documentation)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,36 @@
   <img src="Substra-logo-color.svg" width="600"/>
 </div>
 
-Substra is a python library and a command line interface to interact with the Substra platform.
-For more information on what is substra and how to use it, please refer to the [user documentation](https://docs.substra.org/).
+
+Substra is an open source federated learning (FL) software. It enables the training and validation of machine learning models on distributed datasets. It provides a flexible Python interface and a web application to run federated learning training at scale. This specifc reposity is the low-level Python library used to interact with a Substra network.
+
+Substra's main usage is in production environments. It has already been deployed and used by hospitals and biotech companies (see the MELLODDY project for instance). Substra can also be used on a single machine to perform FL simulations and debug code.
```
Comment (ThibaultFy) [inline README.md:12]: Do you have a link for `MELLODDY` ?

File: README.md [hunk 2]
```diff
@@ -6,13 +6,36 @@
...
+
+Try out our [MNIST example](https://docs.substra.org/en/stable/substrafl_doc/examples/index.html#example-to-get-started-using-the-pytorch-interface).
+
+## Support
+
+If you need support, please either raise an issue on Github or ask on [Slack](https://join.slack.com/t/substra-workspace/shared_invite/zt-1fqnk0nw6-xoPwuLJ8dAPXThfyldX8yA).
+
+## Real world use cases
+
+Discover how Substra powers projects solving critical scientific questions in a privacy-preserving way. Read more here.
```
Comment (ThibaultFy) [inline README.md:31]: Need a link for `here`

**Review by ThibaultFy — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (RomainGoussault) [reply README.md]: removed it for now

File: README.md (hunk 1 above)
Comment (RomainGoussault) [reply README.md]: added

**Review by ThibaultFy — APPROVED** [review]

**Pull request merged**